src/core/Akka.Persistence/Eventsourced.Recovery.cs

```diff
@@ -208,8 +208,9 @@ private EventsourcedState Recovering(Receive recoveryBehavior, TimeSpan timeout)
                         case RecoverySuccess success:
                             timeoutCancelable.Cancel();
                             OnReplaySuccess();
-                            _sequenceNr = success.HighestSequenceNr;
-                            LastSequenceNr = success.HighestSequenceNr;
+                            var highestSeqNr = Math.Max(success.HighestSequenceNr, LastSequenceNr);
```

Comment:
LGTM

Reply:
These changes are making the tests for `Akka.Persistence.Oracle` to fail. Reverting them fixes it. 

The original PR also had a `SnapshotRecoveryWithEmptyJournalSpec` test that we're missing and could shed some light.

Reply:
Nevermind, it was a problem with the Oracle plugin itself.